src/coq/Handlers/Serialization.v

```
@@ -141,12 +141,42 @@ Module Type SerializationBase (LP : LLVMParams) (MP : MemoryParams LP).
               | None => extractbytes_to_dvalue M undef_handler ERR_M lift_ue bytes dt
               end
             else extractbytes_to_dvalue M undef_handler ERR_M lift_ue bytes dt
+        | UVALUE_InsertValue t uv elt idxs =>
+            str <- concretize_uvalueM M undef_handler ERR_M lift_ue uv;;
+            elt <- concretize_uvalueM M undef_handler ERR_M lift_ue elt;;
+            let fix loop str idxs : ERR_M dvalue :=
+              match idxs with
+              | [] => ret str (* What to do? Do we print out some error *)
+              | i :: nil =>
+                  v <- insert_into_str str elt i;;
+                  ret v
+              | i :: tl =>
+                      v <- index_into_str_dv str i;;
+                      v2 <- loop v tl;;
+                      v3 <- insert_into_str str v2 i;;
+                      ret v3
+              end in
+            lift_ue dvalue (loop str idxs)
+        | UVALUE_ExtractElement vec_typ vec idx =>
+            dvec <- concretize_uvalueM M undef_handler ERR_M lift_ue vec;;
+            didx <- concretize_uvalueM M undef_handler ERR_M lift_ue idx;;
+            let elt_typ := match vec_typ with
```

Made another comment, but let's raise an error here instead of introducing a garbage value that might break things in unexpected ways.